Tasks/ExtractFiles/extractFiles.ts

```diff
@@ -0,0 +1,307 @@
+            for (var j = 0; j < matched.length; j++) {
+                var match = path.normalize(matched[j]);
+                if (fs.lstatSync(match).isFile()) {
+                    if (matchingFilesSet.add(match)){
+                        tl.debug('adding file: ' + match);
+                    }
+                }
+            }
+        }
+    }
+
+    return matchingFilesSet.toArray();
+}
+
+function parsePattern(normalizedPattern: string): { file: string, directory: string, search: string }  {
+    tl.debug('parsePattern: ' + normalizedPattern);
+
+    // the first occurance of a wild card, * or ?
+    var firstWildIndex = normalizedPattern.indexOf('*');
+    var questionIndex = normalizedPattern.indexOf('?');
+    if (questionIndex > -1 && (firstWildIndex == -1 || questionIndex < firstWildIndex)) {
+        firstWildIndex = questionIndex;
+    }
+
+    // no wildcards
+    if (firstWildIndex == -1) {
+        return {
+            file: makeAbsolute(normalizedPattern),
+            directory: null,
+            search: null
+        };
+    }
+
+    // search backwards from the first wild card char for the nearest path separator
+    for (var i = firstWildIndex - 1; i > -1; i--) {
+        if (normalizedPattern.charAt(i) == path.sep) {
+            return {
+                file: null,
+                directory: makeAbsolute(normalizedPattern.substring(0, i + 1)),
+                search: normalizedPattern.substring(i + 1, normalizedPattern.length)
+            };
+        }
+    }
+
+    console.log('No path specified for search pattern: ' + normalizedPattern + ' defaulting to: ' + repoRoot);
+
+    return {
+        file: null,
+        directory: repoRoot,
+        search: normalizedPattern
+    };
+}
+
+function makeAbsolute(normalizedPath: string): string {
+    tl.debug('makeAbsolute:' + normalizedPath);
+
+    var result = normalizedPath;
+    if (!path.isAbsolute(normalizedPath)) {
+        result = repoRoot + path.sep + normalizedPath;
+        console.log('Relative file path: '+ normalizedPath+' resolving to: ' + result);
+    }
+    return result;
+}
+
+// Find matching archive files
+var files: string [] = findFiles();
+console.log('Found: ' + files.length + ' files to extract:');
+for (var i = 0; i < files.length; i++) {
+    console.log(files[i]);
+}
+
+// Clean the destination folder before extraction?
+if (cleanDestinationFolder && tl.exist(destinationFolder)) {
+    console.log('Cleaning destination folder before extraction: ' + destinationFolder);
+    tl.rmRF(destinationFolder, false);
+}
+    
+// Create the destination folder if it doesn't exist
+if (!tl.exist(destinationFolder)) {
+    console.log('Creating destination folder: ' + destinationFolder);
+    tl.mkdirP(destinationFolder);
+}
+
+function isZip(file) {
+    return file.endsWith('.zip')
+        || file.endsWith('.jar')
+        || file.endsWith('.war')
+        || file.endsWith('.ear');
```

Comment:
Can you add .ipa and .appx and .apk here so that we support unzipping these mobile archive formats?